### PR TITLE
Fix: selection in gantt chart is not surrounding event click(fixes #170)

### LINKED ISSUE #170: Selection in Gantt is not surrounding Event clicked
This is what happens on implementation of gantt:

image

This was the older bug logged a while ago: https://github.com/cerner/carbon-graphs/issues/115

### PR DESCRIPTION
**Pull request by AshishMotanamGurunadham**
**What is the purpose of this pull request? (Check any that's applicable)**

-   [ ] Feature
-   [x] Bug fix
-   [ ] Dependency updates
-   [ ] Documentation updates
-   [ ] Build related changes
-   [ ] Changes an existing functionality
-   [ ] Other, please explain:

**Issue**:
https://github.com/cerner/carbon-graphs/issues/170

**Does this introduce a breaking change?**

-   [ ] Yes
-   [x] No

**What changes did you make? (Give an overview)**

- Added new method to generate selection circle for events to resolve the issue.

**Add any screenshots**

**Before:**

image

image

**After:**

image

image

### CODE REVIEW

**Comment by Veera81** [comment]
Can you resolve the conflicts?

File: src/main/js/helpers/shapeSVG.js [hunk 1]
```diff
@@ -15,3 +15,24 @@ export const getSVGObject = (shape, { posX, posY, scale }) => {
     d3Shape.options.scale = d3Shape.options.scale * scale;
     return d3Shape;
 };
+
+/**
+ * Returns an object to create a svg circle for event data point
+ *
+ * @private
```
Comment (Veera81) [inline src/main/js/helpers/shapeSVG.js:22]: nit: This is not a private method. We are importing in other file and using

File: src/main/js/helpers/shapeSVG.js [hunk 2]
```diff
@@ -15,3 +15,24 @@ export const getSVGObject = (shape, { posX, posY, scale }) => {
...
+ * @param {object} shape - SHAPE object
+ * @param {number} eventShape - properties of event
+ * @param {number} scaleFactor - Scale factor for datapoint
```
Comment (Veera81) [inline src/main/js/helpers/shapeSVG.js:25]: I don't see scaleFactor as parameter name here: A third parameter is just an object which as posX, posY and scale

File: src/main/js/controls/Gantt/helpers/datapointHelpers.js [hunk 3]
```diff
@@ -82,6 +82,49 @@ export const renderSelectionPath = (scale, config, path, dataPoint, index) =>
         )
     );
 
+/**
+ * Renders the circle svg element for event which shows up when clicked on the data point.
+ * It is hidden by default and toggled visible onClick.
+ *
+ * @private
```
Comment (Veera81) [inline src/main/js/controls/Gantt/helpers/datapointHelpers.js:89]: nit: This is also not a private method

**Comment by Veera81** [comment]
Can you resolve the conflicts?

File: src/main/js/controls/Gantt/helpers/datapointHelpers.js (hunk 3 above)
Comment (AshishMotanamGurunadham) [reply src/main/js/controls/Gantt/helpers/datapointHelpers.js]: [fixed](https://github.com/cerner/carbon-graphs/pull/219/commits/33b5cca47796ca87966a1c265591cd6cc253cb5a#diff-890519d63508b6cb8fa8894a8026b166R88)

File: src/main/js/helpers/shapeSVG.js (hunk 1 above)
Comment (AshishMotanamGurunadham) [reply src/main/js/helpers/shapeSVG.js]: [fixed](https://github.com/cerner/carbon-graphs/pull/219/commits/33b5cca47796ca87966a1c265591cd6cc253cb5a#diff-804709ab7db5d3fcf4b59f59f943c28dR20)

**Pull request merged**